recipes/ucc/meta.yaml

```diff
@@ -0,0 +1,54 @@
+package:
+  name: ucc
+  version: v1.2.0
+
+source:
+  url: https://github.com/openucx/ucc/archive/refs/tags/v1.2.0.tar.gz
+  sha256: c1552797600835c0cf401b82dc89c4d27d5717f4fb805d41daca8e19f65e509d
+  patches:
+    - 0001-EC-CUDA-fix-build-with-12.2-and-older-arch-847.patch  # [(cuda_compiler_version or "None").startswith("12")]
+
+build:
+  skip: True  # [(not linux) or (cuda_compiler_version == 'None')]
+  number: 0
+  missing_dso_whitelist:
+    - '*libcuda.so*'
+  ignore_run_exports_from:
+    - cuda-nvcc
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - automake
+    - libtool
+    - make
+
+  host:
+    - ucx
+    - cuda-cudart-dev ={{ cuda_compiler_version }}  # [(cuda_compiler_version or "None").startswith("12")]
+
+  run:
+    - ucx
+
+  run_constrained:
+    - cuda-version >=11.2,<12  # [(cuda_compiler_version or "None").startswith("11")]
+    - cuda-version >=12,<13    # [(cuda_compiler_version or "None").startswith("12")]
```

Comment:
This should not be necessary because our CUDA compiler packages already have strong run_exports on cuda-version for similar pinnings.

```suggestion
```